FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -143,6 +141,38 @@ + (void)validateAppOptions:(FIROptions *)appOptions appName:(NSString *)appName
             kFIRLoggerInstallations, kFIRInstallationsMessageCodeInvalidFirebaseAppOptions,
             [missingFields componentsJoinedByString:@", "]];
   }
+
+  [self validateAPIKey:appOptions.APIKey];
+}
+
++ (void)validateAPIKey:(nullable NSString *)APIKey {
+  NSMutableArray<NSString *> *validationIssues = [NSMutableArray array];
```

Comment:
nit: Internally I think alloc] init] is prefered?

Reply:
done